src/Thinktecture.Relay.Server/Middleware/RelayMiddleware.cs

```diff
@@ -314,4 +323,31 @@ private async Task<bool> TryInlineBodyContentAsync(TRequest request, Cancellatio
 		CancellationToken cancellationToken)
 		=> _responseWriter.WriteAsync(_relayContext.ClientRequest.CreateResponse<TResponse>(httpStatusCode), response,
 			cancellationToken);
+
+	private async Task<Guid?> LoadTenantByIdAsync(string name, CancellationToken cancellationToken = default)
+	{
+		var normalizedName = _tenantService.NormalizeName(name);
+		var cacheKey = $"{normalizedName}_id";
+
+		var cachedId = await _cache.GetAsync(cacheKey, cancellationToken);
+		if (cachedId != null)
+		{
+			return new Guid(cachedId);
+		}
+
+		var tenant = await _tenantService.LoadTenantByNameAsync(name);
+		if (tenant == null)
+		{
+			return null;
+		}
+
+		var tenantId = tenant.Id;
+		cachedId = tenantId.ToByteArray();
+		await _cache.SetAsync(cacheKey,
+			cachedId,
+			new DistributedCacheEntryOptions().SetAbsoluteExpiration(_options.TenantIdCacheTimeout),
```

Comment:
If this is moved to a local variable the line may be shorter and in one line only.